composer/callbacks/system_metrics_monitor.py

```diff
@@ -46,9 +90,25 @@ def run_event(self, event: Event, state: State, logger: Logger):
         ]:
             local_node_system_metrics = self.compute_system_metrics()
             all_system_metrics = dist.all_gather_object(local_node_system_metrics)
-            system_metrics = {
-                key: value for local_metrics in all_system_metrics for key, value in local_metrics.items()
-            }
+            system_metrics = {}
+            gpu_metrics_set = set(_GPU_METRICS)
+
+            if self.log_all_data:
+                for rank, metrics in enumerate(all_system_metrics):
+                    for key, value in metrics.items():
+                        if key in gpu_metrics_set:
+                            system_metrics[f'Rank_{rank}_{key}'] = value
+                        else:
+                            system_metrics[key] = value
+
+            else:
+                model_device = next(state.model.parameters()).device
+                system_metrics = self.compute_min_max_metrics(all_system_metrics, model_device)
+                for rank, metrics in enumerate(all_system_metrics):
+                    for key, value in metrics.items():
+                        if key not in gpu_metrics_set:
+                            system_metrics[key] = value
```

Comment:
why is this if statement needed?

Reply:
we need to still log the cpu metrics without logging the original gpu metrics without the added prefixes